### PR TITLE
Add UI settings to Map components

### PR DESCRIPTION
**Pull request by aprilzhang**

### CODE REVIEW

File: src/lib/components/DeckGLMap/components/settings/Settings.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            alignItems: "center",
+        },
+    })
+);
+const Settings: React.FC = React.memo(() => {
+    const classes = useStyles();
+
+    const spec = useSelector((st: MapState) => st.spec);
+
+    if (isEmpty(spec) || !("layers" in spec)) return null;
```
Comment (alexandruandrei91) [inline src/lib/components/DeckGLMap/components/settings/Settings.tsx:30]: The spec should either be `null` or valid (non-empty, containing `"layers"`), so probably this check should just be for a `null` spec.

File: src/lib/components/DeckGLMap/utils/specExtractor.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { isEmpty } from "lodash";
+import { DrawMode } from "../redux/types";
+
+export const getLayerVisibility = (
+    spec: Record<string, unknown>
+): Record<string, boolean> => {
+    return !isEmpty(spec) && "layers" in spec
```
Comment (alexandruandrei91) [inline src/lib/components/DeckGLMap/utils/specExtractor.ts:8]: Same as above

File: src/lib/components/DeckGLMap/utils/specExtractor.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+              {}
+          )
+        : {};
+};
+
+export const getDrawMode = (
+    spec: Record<string, unknown>,
+    layerId: string
+): DrawMode | null => {
+    if (isEmpty(spec) || !("layers" in spec)) return null;
```
Comment (alexandruandrei91) [inline src/lib/components/DeckGLMap/utils/specExtractor.ts:24]: Same here

File: src/lib/components/DeckGLMap/DeckGLMap.jsx [hunk 4]
```diff
@@ -42,8 +40,7 @@ DeckGLMap.defaultProps = {
 };
 
 function DeckGLMap({ id, resources, deckglSpecPatch, coords, setProps }) {
-    const [deckglSpec, setDeckglSpec] = React.useState(null);
-
+    const [deckglSpec, setDeckglSpec] = React.useState({});
```
[the remark this answers is not in the document]
Comment (aprilzhang) [reply src/lib/components/DeckGLMap/DeckGLMap.jsx]: Thank you, done

**Review by alexandruandrei91 — APPROVED** [review]
It looks great 👍

**Pull request merged**